src/frontend/src/storage/index.ts

```diff
@@ -156,6 +180,32 @@ const migrated = (): Anchors | undefined => {
   return anchors;
 };
 
+/** "Low-level" functions to read anchors from and write anchors to IndexedDB */
+
+const readIndexedDB = async (): Promise<Anchors | undefined> => {
+  const item: unknown = await idb.get("anchors");
+
+  if (item === undefined) {
+    return;
+  }
+
+  // Read the object
+  const parsed = Anchors.safeParse(item);
+  if (parsed.success !== true) {
+    const message =
+      `could not read saved identities: ignoring malformed IndexedDB data: ` +
+      parsed.error;
+    console.warn(message);
+    return {};
+  }
+
+  return parsed.data;
+};
+
+const writeIndexedDB = async (anchors: Anchors) => {
+  await idb.set("anchors", anchors);
```

Comment:
nth: you can spare the "await" here I think

Reply:
Yeah, just wanted to be consistent with the other functions. Don't feel strongly about it.

Reply:
Just to be clear, the `await` would be replaced with a `return` otherwise the promise will be lost

Reply:
```
const writeIndexedDB = (anchors: Anchors): Promise<void> => set("anchors", anchors);
```

But doesn't matter that much